tests/plugins/test_generate_maven_metadata.py

```diff
@@ -0,0 +1,303 @@
+
+    def mavenfile(self, maveninfo):
+        return '{group_id}/{artifact_id}/{version}/{filename}'.format(**maveninfo)
+
+
+def mock_koji_session(koji_proxyuser=None, koji_ssl_certs_dir=None,
+                      koji_krb_principal=None, koji_krb_keytab=None,
+                      build_info=None, archives=None):
+    if not build_info:
+        build_info = DEFAULT_KOJI_BUILD
+    if not archives:
+        archives = DEFAULT_ARCHIVES
+
+    flexmock(koji, PathInfo=MockedPathInfo)
+
+    session = flexmock()
+
+    (flexmock(koji)
+     .should_receive('ClientSession')
+     .once()
+     .and_return(session))
+
+    def mock_get_build(nvr):
+        if nvr == DEFAULT_KOJI_BUILD['nvr']:
+            return DEFAULT_KOJI_BUILD
+        else:
+            return None
+
+    (session
+     .should_receive('getBuild')
+     .replace_with(mock_get_build))
+
+    (session
+     .should_receive('listArchives')
+     .and_return(archives))
+
+    (session
+     .should_receive('krb_login')
+     .and_return(True))
+    return session
+
+
+def mock_workflow(tmpdir):
+    workflow = DockerBuildWorkflow(source=MOCK_SOURCE)
+    mock_source = MockSource(tmpdir)
+    setattr(workflow, 'builder', X)
+    workflow.builder.source = mock_source
+    flexmock(workflow, source=mock_source)
+    mock_get_retry_session()
+
+    return workflow
+
+
+def mock_fetch_artifacts_by_nvr(tmpdir, contents=None):
+    if contents is None:
+        contents = dedent("""\
+            - nvr: com.sun.xml.bind.mvn-jaxb-parent-2.2.11.4-1
+            """)
+
+    with open(os.path.join(tmpdir, REPO_FETCH_ARTIFACTS_KOJI), 'w') as f:
+        f.write(contents)
+        f.flush()
+
+
+def mock_nvr_downloads(build_info=None, archives=None, overrides=None):
+    if not build_info:
+        build_info = DEFAULT_KOJI_BUILD
+    if not archives:
+        archives = DEFAULT_ARCHIVES
+    if not overrides:
+        overrides = {}
+
+    pi = koji.PathInfo(topdir=KOJI_ROOT)
+
+    for archive in archives:
+        url = pi.mavenbuild(build_info) + '/' + pi.mavenfile(archive)
+        # Use any overrides for this archive ID
+        archive_overrides = overrides.get(archive['id'], {})
+        status = archive_overrides.get('status', 200)
+        body = archive_overrides.get('body', archive['filename'] + archive['group_id'])
+        responses.add(responses.GET, url, body=body, status=status)
+
+
+def make_and_store_reactor_config_map(workflow, additional=None):
+    workflow.plugin_workspace[ReactorConfigPlugin.key] = {}
+
+    reactor_map = {
+        'version': 1,
+        'koji': {
+            'hub_url': KOJI_HUB,
+            'root_url': KOJI_ROOT,
+            'auth': {}
+        }
+    }
+    if additional:
+        reactor_map.update(additional)
+
+    workflow.plugin_workspace[ReactorConfigPlugin.key] = {
+        WORKSPACE_CONF_KEY: ReactorConfig(reactor_map)
+    }
+
+
+@responses.activate  # noqa
+def test_generate_maven_metadata(tmpdir, docker_tasker, user_params):
```

Comment:
Do you think that would be possible to utilize this mocking here?
https://github.com/containerbuildsystem/atomic-reactor/blob/master/tests/mock_env.py

Reply:
Updated to use mock_env.